### PR TITLE
Add wan2.2-distill (lightx2v) , Index-AniSora V3.2 and LoRA adapter pipeline for Wan2.2 I2V. 

### PR DESCRIPTION
**Pull request by tvardhineniTT**
### Summary

This PR adds three new Wan2.2 I2V pipeline variants, all validated end-to-end on BH Galaxy 4×8 Ring:

     a.  Wan2.2-Distill (lightx2v) — 4-step distilled I2V with CFG baked in. 6.8–7.3× speedup over base 40-step pipeline (7.96 s @ 480p, 20.0 s @ 720p).
     b.  Index-AniSora V3.2 — Anime-domain fine-tune of Wan2.2 I2V-14B. Full checkpoint replacement (not LoRA/distilled). Configurable step count (tested at 40, 16, 8 steps).
     c.  LoRA adapter pipeline — CPU-side LoRA fusion into base Wan2.2 weights before device push. Supports kohya/A1111 and diffusers-style key formats, split and single-file LoRA modes. Demonstrated with camera control LoRAs (earth zoom-out, camera rotation).
All three reuse the existing WanPipelineI2V base class and share the base Wan2.2 tokenizer, UMT5 encoder, VAE, and scheduler. No modifications in tt_dit. 

### Wan2.2 distill lightx2v model performance : 
**Headline numbers (BH Galaxy 4×8 Ring, 81 frames, today's run):**                                                                                                       
                                                                                                                                                                           
  | Resolution | Base i2v (40 steps, CFG) | Distill (4 steps, no CFG) | Speedup |                                                                                          
  |---|---:|---:|---:|                                                                                                                                                     
  | 480p (832×480) | 54.4 s | **8.0 s** | **6.84×** |                                                                                                                      
  | 720p (1280×720) | 147.1 s | **20.0 s** | **7.34×** |          
  
### LoRA pipeline : 
Earth Zoomout LoRA  results : same dog image, same prompt. 
Original NO LoRA: 

https://github.com/user-attachments/assets/50e6421e-7227-4c9a-83b6-ead16d0c6078


With LoRA : 

https://github.com/user-attachments/assets/aca7dbcd-3102-4dab-851e-93e8d0f88111

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [![](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml/badge.svg?branch=wan2.2_distil_anisora_lora)](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:wan2.2_distil_anisora_lora)
- [![](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml/badge.svg?branch=wan2.2_distil_anisora_lora)](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:wan2.2_distil_anisora_lora)
- [![](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml/badge.svg?branch=wan2.2_distil_anisora_lora)](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:wan2.2_distil_anisora_lora)
- [![](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml/badge.svg?branch=wan2.2_distil_anisora_lora)](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:wan2.2_distil_anisora_lora)
- [![](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml/badge.svg?branch=wan2.2_distil_anisora_lora)](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:wan2.2_distil_anisora_lora)
- [![](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml/badge.svg?branch=wan2.2_distil_anisora_lora)](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:wan2.2_distil_anisora_lora)
- [![](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml/badge.svg?branch=wan2.2_distil_anisora_lora)](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:wan2.2_distil_anisora_lora)
- [![](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml/badge.svg?branch=wan2.2_distil_anisora_lora)](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:wan2.2_distil_anisora_lora)

### CODE REVIEW

File: models/tt_dit/tests/models/wan2_2/test_performance_wan.py [hunk 1]
```diff
@@ -107,14 +107,20 @@ def i2v_metrics(mesh_device, height):
 
 
 def wan_pipeline_metrics_condimg(mesh_device, width, height, model_type, topology: ttnn.Topology):
+    import os
```
Comment (sosborne-TT) [inline models/tt_dit/tests/models/wan2_2/test_performance_wan.py]: nit, move import to top of the file

File: models/tt_dit/tests/models/wan2_2/test_performance_wan.py [hunk 2]
```diff
@@ -212,8 +218,12 @@ def test_pipeline_performance(
     tp_factor = tuple(mesh_device.shape)[tp_axis]
 
     # Test prompts
+    import os as _os
```
Comment (sosborne-TT) [inline models/tt_dit/tests/models/wan2_2/test_performance_wan.py]: nit move import to top of file

**Review by sosborne-TT — APPROVED** [review]

**Comment by tvardhineniTT** [comment]
@sadesoyeTT  Can you please review this PR.

File: models/tt_dit/tests/models/wan2_2/test_performance_wan.py [hunk 3]
```diff
@@ -114,7 +115,11 @@ def wan_pipeline_metrics_condimg(mesh_device, width, height, model_type, topolog
     else:
         pipeline_cls = WanPipelineI2V
         expected_metrics = i2v_metrics(mesh_device, height)
-        image_prompt = create_fractal_image(width, height)
+        prompt_image_path = os.environ.get("PROMPT_IMAGE")
```
Comment (sadesoyeTT) [inline models/tt_dit/tests/models/wan2_2/test_performance_wan.py]: Can you move the custom image and text prompt testing to the pipeline test.

File: models/tt_dit/tests/models/wan2_2/test_pipeline_lora.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        is_fsdp=is_fsdp,
+        height=height,
+        width=width,
+        num_frames=num_frames,
+        lora_high_path=os.environ["LORA_HIGH_PATH"],
+        lora_low_path=os.environ.get("LORA_LOW_PATH"),
+        lora_scale=lora_scale,
+    )
+
+    prompt = os.environ.get("PROMPT") or "A golden retriever running on a sandy beach, waves in the background"
```
Comment (sadesoyeTT) [inline models/tt_dit/tests/models/wan2_2/test_pipeline_lora.py]: We should use the same convention. i.e NO_PROMPT should use the default prompt you have i.e "A golden ret....". For custom prompt, user should use the keyboard input you have below.

File: models/tt_dit/tests/models/wan2_2/test_pipeline_lora.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+            run(prompt=prompt, number=0, seed=42)
+        else:
+            for i in itertools.count():
+                new_prompt = input("Enter the input prompt, or q to exit: ")
+                if new_prompt:
+                    prompt = new_prompt
+                if prompt[0] == "q":
+                    break
+                run(prompt=prompt, number=i, seed=i)
+    finally:
```
Comment (sadesoyeTT) [inline models/tt_dit/tests/models/wan2_2/test_pipeline_lora.py]: I am not sure what is going on here. We should ideally remove this and use the parameters of the pipeline.

File: models/tt_dit/tests/models/wan2_2/test_pipeline_wan_distill_i2v.py [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+        num_links=num_links,
+        dynamic_load=dynamic_load,
+        topology=topology,
+        is_fsdp=is_fsdp,
+        height=height,
+        width=width,
+        num_frames=num_frames,
+    )
+
+    prompt = os.environ.get("PROMPT") or "The cat in the hat runs up the hill to the house."
```
Comment (sadesoyeTT) [inline models/tt_dit/tests/models/wan2_2/test_pipeline_wan_distill_i2v.py]: same NO_PROMPT comment above

File: models/tt_dit/experimental/utils/lightx2v_loader.py [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+``time_embedding.0.weight``, ...), but tt_dit's ``WanTransformer3DModel``
+consumes the *diffusers* layout (``blocks.X.attn1.to_q.weight``,
+``blocks.X.scale_shift_table``, ``proj_out.weight``,
+``condition_embedder.time_embedder.linear_1.weight``, ...).
+
+:func:`load_lightx2v_state_dict` applies :func:`wan_lightx2v_to_diffusers_key`
+to every key by default so the returned dict matches what
+``WanTransformer3DModel._prepare_torch_state`` expects.
+"""
+from __future__ import annotations
```
Comment (sadesoyeTT) [inline models/tt_dit/experimental/utils/lightx2v_loader.py:21]: Can you look through the DiT library to see how we do remapping. It is very close to what you have, but will remove code duplicates. Example [here](https://github.com/tenstorrent/tt-metal/blob/main/models/tt_dit/models/transformers/transformer_motif.py#L59).

File: models/tt_dit/pipelines/wan/pipeline_anisora.py [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+            ),
+        )
+
+    @staticmethod
+    def create_pipeline(*args, random_weights: bool | None = None, **kwargs):
+        kwargs["checkpoint_name"] = kwargs.get("checkpoint_name") or AniSoraPipeline.BASE_DIFFUSERS_REPO
+        if "allow_download" in kwargs:
+            allow = kwargs.pop("allow_download")
+            if allow:
+                os.environ["TT_DIT_ALLOW_HF_DOWNLOAD"] = "1"
```
Comment (sadesoyeTT) [inline models/tt_dit/pipelines/wan/pipeline_anisora.py]: We should restrict programatic use of envs to reading only. Can we use parameters for this to make the code interaction easier to follow as opposed to setting it envs here.

File: models/tt_dit/pipelines/wan/pipeline_wan_distill.py [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+        if lightx2v_local_dir is None:
+            lightx2v_local_dir = os.environ.get("LIGHTX2V_LOCAL_DIR")
+        if random_weights is None:
+            random_weights = os.environ.get("TT_DIT_RANDOM_WEIGHTS") == "1"
+
+        self._lightx2v_local_dir = lightx2v_local_dir
+        self._allow_download = allow_download
+        self._random_weights = random_weights
+
+        ctx = _patch_torch_transformer_random() if random_weights else contextlib.nullcontext()
```
Comment (sadesoyeTT) [inline models/tt_dit/pipelines/wan/pipeline_wan_distill.py:120]: Any reason to implement it this way?

File: models/tt_dit/pipelines/wan/pipeline_wan_distill.py [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+            ),
+        )
+
+    @staticmethod
+    def create_pipeline(*args, random_weights: bool | None = None, **kwargs):
+        kwargs["checkpoint_name"] = kwargs.get("checkpoint_name") or WanDistillPipelineI2V.BASE_DIFFUSERS_REPO
+        if "allow_download" in kwargs:
+            allow = kwargs.pop("allow_download")
+            if allow:
+                os.environ["TT_DIT_ALLOW_HF_DOWNLOAD"] = "1"
```
Comment (sadesoyeTT) [inline models/tt_dit/pipelines/wan/pipeline_wan_distill.py]: Same comments for setting envs here.

**Review by sadesoyeTT — CHANGES_REQUESTED** [review]

File: models/tt_dit/experimental/utils/lightx2v_loader.py (hunk 7 above)
Comment (tvardhineniTT) [reply models/tt_dit/experimental/utils/lightx2v_loader.py:21]: fixed

**Comment by tvardhineniTT** [comment]
@sadesoyeTT  I added the changes as requested, Please review. Thanks

**Review by mbahnasTT — APPROVED** [review]

**Comment by sadesoyeTT** [comment]
> @sadesoyeTT I added the changes as requested, Please review. Thanks

Sorry for the delay. I added @friedrich (our contractor) to help review as well. In the meantime, can you move your code to tt_dit/experimental (you'll need to create one) to help insulate work from both teams.

**Comment by tvardhineniTT** [comment]
> > @sadesoyeTT I added the changes as requested, Please review. Thanks
> 
> Sorry for the delay. I added @friedrich (our contractor) to help review as well. In the meantime, can you move your code to tt_dit/experimental (you'll need to create one) to help insulate work from both teams.

I moved all the code to tt_dit/experimental @sadesoyeTT . Thanks. 
cc : @mbahnasTT

**Review by sadesoyeTT — APPROVED** [review]

**Comment by tvardhineniTT** [comment]
@sadesoyeTT  Shall I add it to merge queue or wait for @friedrich  to review the changes.
